skimage/segmentation/_felzenszwalb_cy.pyx

```diff
@@ -52,28 +52,28 @@ def _felzenszwalb_cython(image, double scale=1, sigma=0.8,
     # rescale scale to behave like in reference implementation
     scale = float(scale) / 255.
     image = ndi.gaussian_filter(image, sigma=[sigma, sigma, 0])
+    height, width = image.shape[:2]
 
     # compute edge weights in 8 connectivity:
-    down_cost = np.sqrt(np.sum((image[1:, :, :] - image[:-1, :, :])
-    	*(image[1:, :, :] - image[:-1, :, :]), axis=-1))
-    right_cost = np.sqrt(np.sum((image[:, 1:, :] - image[:, :-1, :])
-    	*(image[:, 1:, :] - image[:, :-1, :]), axis=-1))
-    dright_cost = np.sqrt(np.sum((image[1:, 1:, :] - image[:-1, :-1, :])
-	*(image[1:, 1:, :] - image[:-1, :-1, :]), axis=-1))
-    uright_cost = np.sqrt(np.sum((image[1:, :-1, :] - image[:-1, 1:, :])
-    	*(image[1:, :-1, :] - image[:-1, 1:, :]), axis=-1))
+    down_cost = np.sqrt(np.sum((image[1:, :, :] - image[:height-1, :, :]) *
+    	                         (image[1:, :, :] - image[:height-1, :, :]), axis=-1))
```

Comment:
@jni, this kind of indentation makes it clear that the goal was to compute the Frobenius norm. I really wish numpy had this kind of optimization. Maybe this is where pythran comes in!

I'll leave that optimization to an other PR that actually cares to do a benchmark for this.

Reply:
@hmaarrfk if you really want to see optimization, try using `@` here! ;) Although the discontiguity might break it.

Reply:
Na, I listened to @serge-sans-paille's talks, and pythran might be the future